arviz/plots/lmplot.py

```diff
@@ -0,0 +1,331 @@
+"""Plot linear regression figure."""
+import warnings
+from numbers import Integral
+
+import xarray as xr
+import numpy as np
+from xarray.core.dataarray import DataArray
+
+from ..sel_utils import xarray_var_iter
+from ..rcparams import rcParams
+from .plot_utils import default_grid, filter_plotters_list, get_plotting_function
+
+
+def plot_lm(
+    y,
+    idata=None,
+    x=None,
+    y_model=None,
+    y_hat=None,
+    num_pp_samples=50,
+    kind_pp="samples",
+    kind_model="lines",
+    xjitter=False,
+    plot_dim=None,
+    backend=None,
+    y_kwargs=None,
+    y_hat_plot_kwargs=None,
+    y_hat_fill_kwargs=None,
+    y_model_plot_kwargs=None,
+    y_model_fill_kwargs=None,
+    backend_kwargs=None,
+    show=None,
+    figsize=None,
+    textsize=None,
+    axes=None,
+    legend=True,
+    grid=True,
+):
+    """Plot linear regression figure to visualize uncertaininty in mean and ppc values.
+
+    Parameters
+    ----------
+    y : str or DataArray or ndarray
+        If str, variable name from observed_data
+    idata : az.InferenceData object, Optional
+        Optional only if y is Sequence
```

Comment:
sequence is not a valid type according to `y` docstring, so I'm not sure what this means.

Reply:
If any of y, x, y_hat, y_model is str, then idata is necessary. Yea, This line is unclear.

Reply:
This explanation should be the one on the docstring! It's different from the current one too which seeing this looks wrong